### PR TITLE
Add a pull request template with important information for contributors.

### PR DESCRIPTION
**Pull request by pchote**
The goal here is to help contributors help themselves.

See https://github.com/blog/2111-issue-and-pull-request-templates and #13774.

### CODE REVIEW

**Comment by penev92** [comment]
I feel like we need to include notes about extra commits, commit messages and something along the lines of "if you need help/are unsure as to how to proceed, pop into IRC". Otherwise I like the idea.

**Comment by pchote** [comment]
I agree, especially about how commits should be structured.

**Comment by pchote** [comment]
Updated.

**Review by abcdefg30 — COMMENTED** [review]
Looks good to me already. I'd really like it though if we managed to include a link to http://webchat.freenode.net/?channels=openra somehow.

**Comment by penev92** [comment]
"If you need any help you can ask for help"  hmmm...

**Comment by reaperrr** [comment]
> "If you need any help you can ask for help" hmmm...

Just ditching the "for help" should do, in my opinion.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+Thank you for your contribution to OpenRA!
+
+Please be aware that we do not have enough project maintainers to match the rate of contributions, so it may take several days before somebody is able to respond to your Pull Request.
```
Comment (Mailaender) [inline .github/PULL_REQUEST_TEMPLATE.md]: That doesn't sound very welcoming. Why don't we just promote more long-term contributors to maintainer status?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+You can help speed up the review process by following a few steps:
+
+* Make sure that you have read and understand the OpenRA Coding Standard (see https://github.com/OpenRA/OpenRA/wiki/Coding-Standard).
+* Write quality commit messages (see https://chris.beams.io/posts/git-commit/).
+* Only commit changes that directly relate to your Pull Request.  Use your Git interface to unstage any unrelated changes to project files, line endings, or other files.
+* Use the code diff view below to look for (and then fix!) whitespace errors in your code.
```
Comment (Mailaender) [inline .github/PULL_REQUEST_TEMPLATE.md]: The tooling should do this automatically. This should never be a manual process.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+* Use the `make test` and `make check` commands to check for (and fix!) any issues that are reported by our automated tests.
+* If you are changing shared mod or engine code make sure that you have tested your changes in all four default mods.
```
Comment (Mailaender) [inline .github/PULL_REQUEST_TEMPLATE.md]: Hard to read and misses punctuation:

```diff
- If you are changing shared mod or engine code make sure that you have tested your changes in all four default mods.
+ If you are changing shared mod or engine code, make sure that you have tested your changes in all four default mods.
```

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (pchote) [reply .github/PULL_REQUEST_TEMPLATE.md]: This statement is the reality that we have to face.  While we do have enthusiastic newer reviewers like @ltem, @MustaphaTR and @Smittytron helping out where they can, there are many PRs that they do not feel qualified to review.  Most non-trivial PRs sit for multiple weeks waiting for review, and stating this up front shows that this is not a personal snub against the contributor.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (pchote) [reply .github/PULL_REQUEST_TEMPLATE.md]: Once again, this statement reflects the reality of the situation.  Many PRs from new contributors sit with failing tests until a reviewer points them out, and there are other cases (such as tabs vs spaces or project file changes) that are not tested by the automatic tooling.

Comment (Mailaender) [reply .github/PULL_REQUEST_TEMPLATE.md]: In that case I suggest promoting @EditorConfig and #14182 instead.

**Comment by pchote** [comment]
Updated.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+* Write quality commit messages (see https://chris.beams.io/posts/git-commit/).
+* Only commit changes that directly relate to your Pull Request.  Use your Git interface to unstage any unrelated changes to project files, line endings, whitespace, or other files.
+* Review the code diff view below to double check that your changes satisfy the above three points.
+* Use the `make test` and `make check` commands to check for (and fix!) any issues that are reported by our automated tests.
+* If you are changing shared mod or engine code, make sure that you have tested your changes in all four default mods.
+* Respond to review comments as soon as you reasonably can.  Reviewers will usually prioritize Pull Requests that are still fresh in their minds.  Make sure to leave a comment when you push new changes, otherwise GitHub does not automatically notify reviewers!
+* Leave a polite comment asking for reviews if a week or more has passed without feedback.
+
+If you need any help you can in the #openra IRC channel on freenode (most active during European evenings).
```
Comment (reaperrr) [inline .github/PULL_REQUEST_TEMPLATE.md]: This seems to be missing an 'ask' after the 'can' now.

**Review by reaperrr — DISMISSED** [review]

**Comment by reaperrr** [comment]
Looks good to me except that one issue pointed out in the diff.

**Comment by pchote** [comment]
Done.

**Review by reaperrr — APPROVED** [review]

**Review by penev92 — APPROVED** [review]

**Pull request merged**